source/websocket_bootstrap.c

```diff
@@ -117,13 +106,23 @@ int aws_websocket_client_connect(const struct aws_websocket_client_connection_op
     AWS_ASSERT(options);
 
     /* Validate options */
-    if (!options->allocator || !options->bootstrap || !options->socket_options || !options->uri ||
+    struct aws_byte_cursor path;
+    aws_http_request_get_path(options->handshake_request, &path);
+    if (!options->allocator || !options->bootstrap || !options->socket_options || !options->host.len || !path.len ||
         !options->on_connection_setup) {
 
         AWS_LOGF_ERROR(AWS_LS_HTTP_WEBSOCKET_SETUP, "id=static: Missing required websocket connection options.");
         return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
     }
 
+    struct aws_byte_cursor method;
+    aws_http_request_get_method(options->handshake_request, &method);
+    if (!aws_byte_cursor_eq(&method, &s_method_get)) {
```

Comment:
We already have a string->enum function

`if (aws_http_str_to_method(method) != AWS_HTTP_METHOD_GET)`

but we only have a HEAD enum right now, you'll need to create GET